test/e2e/framework/network_pods.go

```diff
@@ -178,6 +188,61 @@ func (np *NetworkPod) CreateService() *v1.Service {
 	return np.framework.CreateTCPService(np.Config.Cluster, np.Pod.Labels[TestAppLabel], np.Config.Port)
 }
 
+// RunCommand run the specified comand in this NetworkPod
+func (np *NetworkPod) RunCommand(cmd []string) (string, string) {
+	req := KubeClients[np.Config.Cluster].CoreV1().RESTClient().Post().
+		Resource("pods").Name(np.Pod.Name).Namespace(np.Pod.Namespace).
+		SubResource("exec").Param("container", np.Config.ContainerName)
+
+	req.VersionedParams(&v1.PodExecOptions{
+		Container: np.Config.ContainerName,
+		Command:   cmd,
+		Stdin:     false,
+		Stdout:    true,
+		Stderr:    true,
+		TTY:       false,
+	}, scheme.ParameterCodec)
+
+	config := RestConfigs[np.Config.Cluster]
+
+	executor, err := remotecommand.NewSPDYExecutor(config, "POST", req.URL())
+	Expect(err).NotTo(HaveOccurred())
+
+	var stdout, stderr bytes.Buffer
+
+	err = executor.Stream(remotecommand.StreamOptions{
+		Stdin:  nil,
+		Stdout: &stdout,
+		Stderr: &stderr,
+		Tty:    false,
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	return stdout.String(), stderr.String()
+}
+
+// GetLog returns container log from this NetworkPod
+func (np *NetworkPod) GetLog() string {
+	req := KubeClients[np.Config.Cluster].CoreV1().RESTClient().Get().
+		Namespace(np.Pod.Namespace).Name(np.Pod.Name).Resource("pods").SubResource("log")
+
+	req.Param("follow", strconv.FormatBool(false)).
+		Param("container", np.Config.ContainerName).
+		Param("previous", strconv.FormatBool(false)).
+		Param("timestamps", strconv.FormatBool(false))
```

Comment:
It seems like you could use 

`KubeClients[np.Config.Cluster].CoreV1().Pods(np.Pod.Namespace).GetLogs(np.Pod.Name, &corev1.PodLogOptions{...})`

Reply:
Good catch. Actually, I also found that I was specifying default values for `.PodLogOptions` (Even `container` isn't necessary, if there is only one). It now become much simple.